addons/sale/static/src/js/product_configurator_mixin.js

```diff
@@ -440,6 +403,51 @@ var ProductConfiguratorMixin = {
     _getProductId: function ($parent) {
         return parseInt($parent.find('.product_id').val());
     },
+    /**
+     * Returns a debounced `_getCombinationInfo` with a leading and a trailing
+     * call, and which is memoized per `uniqueId`.
+     *
+     * The uniqueId is needed because on the configurator modal there might be
+     * multiple elements triggering the rpc at the same time, and we want each
+     * individual product rpc to be executed, but only once per individual
+     * product.
+     *
+     * The leading execution is to keep good reactivity on the first call, for
+     * a better user experience. The trailing is because ultimately only the
+     * information about the last selected combination is useful. All
+     * intermediary rpc can be ignored and are therefore best not done at all.
+     *
+     * @private
+     * @param {string} uniqueId
+     * @returns {function}
+     */
+    _debouncedGetCombinationInfo: _.memoize(function (uniqueId) {
```

Comment:
I can still add an impossible combination to my cart on runbot if I click fast enough:
https://drive.google.com/file/d/13yPu40ucFsNnUBuVMd2NbJblrzMRj_2g/view?usp=sharing

And when I close the modal for that combination the "Add to cart" button is even still available:
https://drive.google.com/file/d/1gBFSoCOj3Awr8ybqCcIw2NuiyOsktDQP/view?usp=sharing

Is there a way to prevent that?

Reply:
1. The product is not yet in cart at that step, it's only added to cart when you click on either of the 2 buttons at the bottom of the modal. and it's at that time the possibility will be validated.

We usually don't write code for edge cases like this. But we can probably prevent it, I can have a look if this is really an issue. 

2. This is strange no matter what we decide. I'll have a look why the disable is not happening (probably because modal open above it).